test/integration/controllermanager/shootstate/shootstate_test.go

```diff
@@ -65,7 +65,10 @@ var _ = Describe("ShootState controller test", func() {
 
 		addFinalizer = func(shootState *gardencorev1beta1.ShootState) {
 			By("Add ShootState finalizer")
-			Expect(controllerutils.AddFinalizers(ctx, testClient, shootState, shootstate.FinalizerName)).To(Succeed())
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(shootState), shootState)).To(Succeed())
+				g.Expect(controllerutils.AddFinalizers(ctx, testClient, shootState, shootstate.FinalizerName)).To(Succeed())
+			}).Should(Succeed())
```

Comment:
See https://github.com/gardener/gardener/blob/63cb5ea67f7e3882851d76db111816ed32b04ec7/docs/development/testing.md#L32

```suggestion
			EventuallyWithOffset(1, func(g Gomega) {
				g.ExpectWithOffset(1, testClient.Get(ctx, client.ObjectKeyFromObject(shootState), shootState)).To(Succeed())
				g.ExpectWithOffset(1, controllerutils.AddFinalizers(ctx, testClient, shootState, shootstate.FinalizerName)).To(Succeed())
			}).Should(Succeed())
```

because the assertions are in a helper func and when it fails, it points to the failure in the helper func and not to the usage in the test. Hence, without the offsets it is not possible to understand which of the ITs using the helper func failed.

Reply:
Thanks for the advice, switched to using the `EventuallyWithOffset`/`ExpectWithOffset` utilities within the `addFinalizer` helper function.